pkg/operator/k8sutil/utils.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sutil
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+)
+
+func okToStopOrContinue(context *clusterd.Context, namespace, deployment, daemon, action string) error {
+	if action == "stop" {
+		err := okToStopDaemon(context, namespace, deployment, daemon, action)
+		if err != nil {
+			return fmt.Errorf("failed to check if %s was ok to %s", deployment, action)
+		}
+	}
+	if action == "continue" {
+		// the mon case is handled directly in the deployment where the mon checks for quorum
+		switch daemon {
+		case "osd":
+			err := okToContinueOSDDaemon(context, namespace)
+			if err != nil {
+				return fmt.Errorf("failed to check if %s was ok to %s", deployment, action)
+			}
+		case "mds":
+			err := okToContinueMDSDaemon(context, namespace, deployment, daemon, action)
+			if err != nil {
+				return fmt.Errorf("failed to check if %s was ok to %s", deployment, action)
+			}
+		}
+	}
+
+	return nil
+}
+
+func okToStopDaemon(context *clusterd.Context, namespace, deployment, daemon, action string) error {
+	deploymentSplit := strings.Split(deployment, "-")
+	daemonID := deploymentSplit[len(deploymentSplit)-1]
+
+	args := []string{daemon, "ok-to-stop", daemonID}
+	buf, err := client.ExecuteCephCommandPlain(context, namespace, args)
```

Comment:
The `ok-to-stop` command isn't going to be available until the next ceph releases, right? Do we need to check if this flag exists before we try to use it?

Reply:
Yes yes, hence the draft :-).
Currently, it exists for OSD but MON and MDS are new.